[docs] Add a guide for setting dark mode by default (#34839)
mui/material-ui: docs/data/joy/customization/dark-mode/dark-mode.md

# Dark mode

> <p class="description">Learn about the different methods for having dark mode with Joy UI.</p>

Reviewer: Not sure:
```suggestion
<p class="description">Learn about the different methods for enabling dark mode with Joy UI.</p>
```

## Set as default

To have your app using dark mode as its default mode, add `defaultMode: 'dark'` into your `<CssVarsProvider>` wrapper component.

{{"demo": "DarkModeByDefault.js"}}

> For server-side applications, check out the framework setup in the [guide](#server-side-rendering) below and use the same value to the `getInitColorSchemeScript` function.

Reviewer 2: ```suggestion
For server-side applications, check out the framework setup in [the section below](#server-side-rendering) and provide the same value to the `getInitColorSchemeScript` function:
```

```js
getInitColorSchemeScript({ defaultMode: 'dark' });
```

> ## Matching device's mode preference

Reviewer: Maybe to make the headers easier to skip read.

```suggestion
## Matching device's preference
```

> To have your app matching the prefferred mode the user has set on their devide, use `defaultMode: 'system'`.

Reviewer: A couple of typos:

```suggestion
Use `defaultMode: 'system'` to have your app match the preferred mode the user has set on their device.
```

Reviewer 2: Iterating on the suggestion above:

```suggestion
Use `defaultMode: 'system'` to set your app's default mode to match the user's chosen preference on their device.
```

```jsx
import { CssVarsProvider } from '@mui/joy/styles';

<CssVarsProvider defaultMode="system">...</CssVarsProvider>;
```

For server-side applications, check out the framework setup in the [guide](#server-side-rendering) above and provide the same value to the `getInitColorSchemeScript` function.

```js
getInitColorSchemeScript({ defaultMode: 'system' });
```

### Identify the system mode

> To check if the user's preference is in light or dark mode, use the `useColorScheme` React hook:

Reviewer: To be more direct?

```suggestion
Use the `useColorScheme` React hook to check if the user's preference is in light or dark mode:
```

```js
import { useColorScheme } from '@mui/joy/styles';

function SomeComponent() {
  const { mode, systemMode } = useColorScheme();
  console.log(mode); // "system"
  console.log(systemMode); // "light" | "dark" based on the user's preference.
}
```

{{"demo": "IdentifySystemMode.js"}}

:::warning

> **Note:** Make sure to use `useColorScheme()` in a component that's inside `<CssVarsProvider>`, otherwise it will throw an error.

Reviewer 2: ```suggestion
The `useColorScheme()` hook only works with components nested inside of `<CssVarsProvider>`—otherwise it will throw an error.
```

> **Note:** Make sure to use `useColorScheme()` in a component that's inside `<CssVarsProvider>`, otherwise it will throw an error.

Reviewer 2: ```suggestion
The `useColorScheme()` hook only works with components nested inside of `<CssVarsProvider>`—otherwise it will throw an error.
```
